webapp/tests/karma/ts/services/xml-forms.service.spec.ts

```diff
@@ -573,18 +572,70 @@ describe('XmlForms service', () => {
             internalId: 'registration',
             _attachments: { xml: { something: true } },
           },
+        },
+        {
+          id: 'form:training:new-feature',
+          doc: {
+            _id: 'form:training:new-feature',
+            internalId: 'new-feature',
+            _attachments: { xml: { something: true } },
+          },
         }
       ];
       dbQuery.resolves({ rows: given });
       UserContact.resolves();
       const service = getService();
-      return service.list({ contactForms: true }).then(actual => {
+      return service.list({ contactForms: true, trainingForms: false }).then(actual => {
         expect(actual.length).to.equal(2);
         expect(actual[0]).to.deep.equal(given[1].doc);
         expect(actual[1]).to.deep.equal(given[2].doc);
       });
     });
 
+    it('should filter for training forms', () => {
+      const given = [
+        {
+          id: 'visit',
+          doc: {
+            _id: 'visit',
+            internalId: 'visit',
+            _attachments: { xml: { something: true } },
+          },
+        },
+        {
+          id: 'form:contact:person',
+          doc: {
+            _id: 'form:contact:person',
+            internalId: 'stock-report',
+            _attachments: { xml: { something: true } },
+          },
+        },
+        {
+          id: 'form:contact:clinic',
+          doc: {
+            _id: 'form:contact:clinic',
+            internalId: 'registration',
+            _attachments: { xml: { something: true } },
+          },
+        },
+        {
+          id: 'form:training:new-feature',
+          doc: {
+            _id: 'form:training:new-feature',
+            internalId: 'new-feature',
+            _attachments: { xml: { something: true } },
+          },
+        }
+      ];
+      dbQuery.resolves({ rows: given });
+      UserContact.resolves();
+      const service = getService();
+      return service.list({ contactForms: false, trainingForms: true }).then(actual => {
+        expect(actual.length).to.equal(1);
+        expect(actual[0]).to.deep.equal(given[3].doc);
+      });
+    });
+
     it('filter for non-contact forms', () => {
```

Comment:
Is this ever used anymore? Do we ever want report forms _and_ training forms?

Reply:
I've changed it a bit to retrieve report forms and training forms, mainly to test that the service can do that, if we ever want it.